core/corehttp/gateway_test.go

```diff
@@ -173,6 +173,71 @@ func TestGatewayGet(t *testing.T) {
 	}
 }
 
+func TestPretty404(t *testing.T) {
+	ns := mockNamesys{}
+	ts, n := newTestServerAndNode(t, ns)
+	defer ts.Close()
+
+	// Create files for test hash
+	_, dagRoot, err := coreunix.AddWrapped(n, strings.NewReader("Custom 404"), "404.html")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	_, dagDeep404, err := coreunix.AddWrapped(n, strings.NewReader("Deep custom 404"), "404.html")
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	dagRoot.(*dag.ProtoNode).AddNodeLink("deeper", dagDeep404)
+	if err != nil {
```

Comment:
theres no error being set here

Reply:
Good point! I've fixed this in `TestIPNSHostnameRedirect` too.